Weekly_Work/Week1/Code_changes-git/Metrics_Root_class.py

```diff
@@ -0,0 +1,202 @@
+            'commit': cdata['Commit'],
+            'commit_date': self._str_to_dt_data(cdata['CommitDate']),
+            'files_no': len(cdata['files']),
+            'refs': cdata['refs'],
+            'parents': cdata['parents'],
+            'files': cdata['files']
+        }
+
+        actions = 0
+        for file in cdata['files']:
+            if 'action' in file:
+                actions += 1
+        cleaned_line['files_action'] = actions
+        if 'Merge' in cdata:
+            cleaned_line['merge'] = True
+        else:
+            cleaned_line['merge'] = False
+
+        return cleaned_line
+            
+    def _clean_issue(self, line):
+        repo_name = line['origin']
+        line_data = line['data']
+        cleaned_line =  \
+        {
+            'repo': repo_name,
+            'hash': line_data['id'],
+            'category': "issue",
+            'author': line_data['user']['login'],
+            'created_date': self._str_to_dt_data(line_data['created_at']),
+            'current_status': line_data['state']   
+        }
+        return cleaned_line
+
+            
+    def _clean_pr(self, line):
+        repo_name = line['origin']
+        line_data = line['data']
+        cleaned_line =  \
+        {
+            'repo': repo_name,
+            'hash': line_data['id'],
+            'category': "pull_request",
+            'author': line_data['user']['login'],
+            'created_date': self._str_to_dt_data(line_data['created_at']),
+            'current_status': line_data['state']   
+        }
+        return cleaned_line
+
+    def _is_source_code(self, commit):
+        """
+        Given a commit structure, which is a dictionary returned by the _summary function, 
+        and given a list of files to exclude using source_code_exclude_list while instantiating 
+        an object, decide whether all the files in a commit are to be excluded or not
+        
+        :param commit: a commit structure, returned by the _summary method.
+        """
+        extension_set = set()
+        for file in commit['files']:
+            extension_set.add(self._get_extension(file))
+
+        if self.source_code_exclude_list is None or len(extension_set.difference(self.source_code_exclude_list)) > 0:
+            return True
+        return False
+    
+    def _get_extension(self, file):
+        """
+        Given a file structure, which is a dictionary and an element of commit['files'], 
+        return the extension of that file. 
+        For files without a standard ".xyz" extension, like LICENCE or AUTHORS, the "others" 
+        extension is used. 
+        
+        :param file: a file structure which is a dictionary, an element of commit["files"]
+        """
+            
+        file_name = file['file']
+        if '.' in file_name:
+            file_extension = file_name.split('.')[1]
+        else:
+            file_extension = "other"
+        return file_extension
+
+    def _str_to_dt_data(self, date):
```

Comment:
This can be either a utility function in the module, or a static function in the root class.

Reply:
Cool